core/common/src/main/java/alluxio/security/authorization/AccessControlList.java

```diff
@@ -155,12 +158,23 @@ public void setEntry(AclEntry entry) {
   /**
    * Checks whether the user has the permission to perform the action.
    *
+   * According to POSIX ACL standard, the permission checking algorithm is:
+   * If the user is the owning user, then the owning user's permission is checked;
+   * Else if a named user entry matches the user, the entry is checked;
+   * Else if groups contain the owning group and the action is permitted by the
+   * owning group, then return true;
+   * Else if a named group entry matches a member of the groups and the entry permits the action,
+   * then return true;
+   * Else if the owning group or any named group entry matches a member of the groups, but no group
+   * permits the action, then return false;
+   * Otherwise, check other's permission.
+   *
    * @param user the user
    * @param groups the groups the user belongs to
    * @param action the action
    * @return whether user has the permission to perform the action
    */
-  public boolean check(String user, List<String> groups, AclAction action) {
+  public boolean checkPermission(String user, List<String> groups, AclAction action) {
```

Comment:
it should be marginally faster because it doesn't need to construct a new instance of AclActions and it doesn't need to merge all AclActions of the groups.

Reply:
I see. They are basically performing the same algorithm, so this could be messy to maintain or fix. I think I would prefer to have them consolidated, but if that is not possible, we should at least leave a comment that the logic is duplicated in 2 methods.